### ISSUE TITLE
Spread in JSX children is removed

### ISSUE THREAD

**Issue by OliverJAsh** [prettier/prettier#3158]
**Prettier 1.7.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuc0DOMAEAPTBeTAHgBMBLANwD5gA6OgbQEYBdAX0IHoyqQAaECAAcYpdMlABDDMgBmEgDZo4-AEYAnCWADWcGAGVBm0lADmyGGoCuykMQhhZCpfwBWabACEN23XokBbOAAZYzhHRRsISxhBaIAmcOcQQzUlNWQQGABPQTg0MDVSYT5kwtgAdVJiGAALZAAOAAZ+JX9ScysbNGMTeTgARUsIeESbGAkVSuq6pDj+CwlSeR6AYQh-fwkMqGgw-kslABUJtFHWViA)
```sh
--parser typescript
```

**Input:**
```tsx
const x = <div>{...[1]}</div>
```

**Output:**
```tsx
const x = <div>{[1]}</div>;

```

**Expected behavior:**
```tsx
const x = <div>{...[1]}</div>
```

**Comment by ikatyang** [comment]
Confirmed on the current master, AST affected.

**Comment by suchipi** [comment]
For others like me wondering "why would you need to do that", see https://github.com/facebook/jsx/issues/57 for history.

**Comment by lydell** [comment]
Luckily, this bug does not seem to be causing any trouble in the wild, because it doesn't seem to be usable in Babel+React: https://codesandbox.io/s/4xw26vynr0

And as far as I understand, the workaround is to simply remove the dots (which looks nicer anyway).

**Comment by suchipi** [comment]
It *could* affect a user who is using TypeScript as their compiler, and is using JSX for non-React, though.

**Comment by azz** [comment]
Fix landed in `1.8.0`
